.gitpod.yml

```diff
@@ -0,0 +1,16 @@
+# This configuration file was automatically generated by Gitpod.
+# Please adjust to your needs (see https://www.gitpod.io/docs/introduction/learn-gitpod/gitpod-yaml)
+# and commit this file to your remote git repository to share the goodness with others.
+
+# Learn more from ready-to-use templates: https://www.gitpod.io/docs/introduction/getting-started/quickstart
+
+tasks:
+  - init: make build image
+github:
+  prebuilds:
+    pullRequestsFromForks: true
+vscode:
+  extensions:
+    - "golang.go"
+    - "GitHub.copilot"
+    - "Grafana.vscode-jsonnet"
```

Comment:
I am not a vscode user, so pardon these stupid questions :) 

Our of curiosity, what happens if someone without copilot opens the pod? 
Also why is "Grafana.vcode-jsonnet" needed? 

I am also not too sure if we should include editor specific configs (maintenance cost) in a repo but I don't have a strong opinions as well :)

Reply:
> I am not a vscode user, so pardon these stupid questions :)

No worries! In case you're using Jetbrain's IDEs, I believe it is possible to automate the installation of plugins (I think they the same as vscode extensions)

https://www.gitpod.io/docs/references/gitpod-yml#jetbrainsproductplugins

> Our of curiosity, what happens if someone without copilot opens the pod?

That's a very good question, I'm not 100% sure, but I believe it will ask for authentication and will just do nothing if access isn't granted

>  Also why is "Grafana.vcode-jsonnet" needed?

We do offer a [jsonnet library](https://github.com/prometheus-operator/prometheus-operator/tree/main/jsonnet) for downstream projects, e.g. kube-prometheus. For sure development of jsonnet is much less frequent than go, but it doesn't hurt to let this extension enabled for when it is needed :)

> I am also not too sure if we should include editor specific configs (maintenance cost) in a repo but I don't have a strong opinions as well :)

That's a fair point, maintaining those lines of code will only benefit Gitpod users. So far, I believe it is only me 😅 but I do invite everyone else to try it out. Once I got used to the workflow, I have never used local development again :P

I don't want to force anything though, it doesn't hurt me much to install the extensions every time I open a workspace and I'm far from being the most frequent reviewer. If people don't use Gitpod, then prebuilds won't save anyone's time